core/src/builder/transaction/txhandler.rs

```diff
@@ -212,19 +219,60 @@ impl<T: State> TxHandler<T> {
         }
     }
 
+    pub fn calculate_all_txins_sighash(
+        &self,
+        needed_entity: EntityType,
+        partial_signature_info: PartialSignatureInfo,
+    ) -> Result<Vec<(TapSighash, SignatureInfo)>, BridgeError> {
+        let mut sighashes = Vec::with_capacity(self.txins.len());
+        for idx in 0..self.txins.len() {
+            let sig_id = self.txins[idx].get_signature_id();
+            let sig_owner = sig_id.get_deposit_sig_owner()?;
+            match sig_owner {
+                DepositSigKeyOwner::Operator(sighash_type) => {
+                    if needed_entity == EntityType::Operator {
```

Comment:
nit: combine into guarded match (ditto for below)
```suggestion
                DepositSigKeyOwner::Operator(sighash_type) if needed_entity == EntityType::Operator => {
```